MOBILE-7013: Introduce ability to set custom host for pbs

KostiantynTselykh opened the pull request:
feat(PrebidRenderingSettings):
- Added ability to change bidServerHost

refactoring(DemoRenderingKotlin)
- Replaced reflection usage for setting bid server host

tests:
- Added unit test to verify that bid server host was changed (if not empty)

KostiantynTselykh committed:
feat(PrebidRenderingSettings):
- Added ability to change bidServerHost

refactoring(DemoRenderingKotlin)
- Replaced reflection usage for setting bid server host

tests:
- Added unit test to verify that bid server host was changed (if not empty)

YuriyVelichkoOpenX commented on PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java:
```diff
@@ -75,6 +73,7 @@
 
     private static SdkInitListener sInitSdkListener;
 
+    private static String sBidServerHost = "https://prebid.openx.net/openrtb2/auction";
```
Need to remove this URL. OpenX is a custom server for Prebid. Can we mimic the host initialization behaviour from Prebid Core?

KostiantynTselykh replied:
Will do

KostiantynTselykh committed:
refactoring(PrebidInternalTestApp):
- Rename prebid rendering demo app module

KostiantynTselykh committed:
feat(PrebidRenderingSettings):
- Introduced Host enum with RUBICON, APPNEXUS and CUSTOM values
- Added ability to set custom host through HOST enum
- Added host url validation to fetchDemand

tests:
- Added unit test to verify fetch demand error on invalid custom host
- Applied change to unit tests that rely on bid server host

KostiantynTselykh commented:
@YuriyVelichkoOpenX Applied changes to mimic prebid core host setup.
Also renamed demo app module to: PrebidInternalTestApp.

YuriyVelichkoOpenX commented on PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/PrebidRenderingSettings.java:
```diff
@@ -91,7 +90,7 @@
     public static void initializeSDK(Context context, final SdkInitListener sdkInitListener)
-    throws AdException {
+            throws AdException {
         Log.d(TAG, "Initializing OpenX SDK");
```
In the scope of future work need to review appearance of the words "OpenX" and "Apollo" one more time and remove them.

KonstantinTselykh replied:
Already in progress in a separate branch.

KostiantynTselykh replied:
Already in progress in a separate branch.

YuriyVelichkoOpenX reviewed (approved):
LGTM
+ short comment
+ make sure that all tests  work
